oom-frontend/src/app/site/pages/auth/signup/signup.component.ts

```diff
@@ -7,9 +13,27 @@ import {Component, OnInit} from '@angular/core';
 })
 export class SignupComponent implements OnInit {
 
-  constructor() { }
+  constructor(private authService: AuthService,
+              private router: Router,
+              private toastr: ToastrService
+  ) { }
+
+  form = new SignUpDTO();
+  registerFail: boolean = false;
 
   ngOnInit(): void {
   }
 
+  submit(): void {
+    this.authService.register(this.form).subscribe(res => {
+        this.toastr.success('Registration successfull.', 'Success!')
+        this.router.navigate(['auth/signin']);
+      },
+      err => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Out-Of-Memory-Team/out-of-memory/pullRequest?prid=7533964)
```suggestion
      (err) => {
```